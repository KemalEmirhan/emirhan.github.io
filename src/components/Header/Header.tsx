import * as React from 'react';
import Link from 'next/link';

const Header = (): React.ReactElement => {
    return (
        <div>
            <Link href="/">
                <a title="Ana Sayfa" id="ana-sayfa">                
                    <h3>emrhnksm.</h3>
                </a>
            </Link>
        </div>
    );
}

export default Header;
