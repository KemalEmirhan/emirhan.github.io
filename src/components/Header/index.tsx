'use client';

import React from 'react';
import Avatar from '@/components/Avatar';

const Header = () => {
  const date = new Date().toLocaleDateString('en-US', {
    year: 'numeric',
    month: 'long',
    day: 'numeric',
  });

  return (
    <header className='w-full md:max-w-5xl mx-auto'>
      <div className='grid md:grid-cols-[145px_1fr] items-center py-6 px-4 lg:px-0 md:py-8 gap-6'>
        <div className='hidden md:block border-r border-black border-dashed pr-4 h-full'>
          <Avatar />
        </div>
        <div className='flex items-center justify-between'>
          <h1 className='text-2xl font-bold'>emrhnkml.</h1>
          <time dateTime={date}>{date}</time>
        </div>
      </div>
    </header>
  );
};

export default Header;
