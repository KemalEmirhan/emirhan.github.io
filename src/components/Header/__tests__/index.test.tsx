import { render, screen } from '@testing-library/react';
import {
  afterAll,
  beforeAll,
  beforeEach,
  describe,
  expect,
  it,
  vi,
} from 'vitest';
import Header from '../';

describe('Header', () => {
  beforeAll(() => vi.useFakeTimers());
  beforeEach(() => vi.setSystemTime(new Date('2025-01-01T00:00:00Z')));
  afterAll(() => vi.useRealTimers());

  it('should render', () => {
    const { container } = render(<Header />);
    expect(container).toMatchSnapshot();
  });

  it('should render the main heading', () => {
    render(<Header />);

    const heading = screen.getByRole('heading', { level: 1 });
    expect(heading).toHaveTextContent('emrhnkml.');
  });

  it('should render the current date', () => {
    render(<Header />);

    const timeElement = screen.getByText(/^\w+ \d{1,2}, \d{4}$/);
    expect(timeElement).toBeInTheDocument();
    expect(timeElement.tagName).toBe('TIME');

    // Check that the date is in the expected format
    const dateText = timeElement.textContent;
    expect(dateText).toMatch(/^\w+ \d{1,2}, \d{4}$/);
  });

  it('should have proper accessibility attributes', () => {
    render(<Header />);

    const header = screen.getByRole('banner');
    expect(header).toBeInTheDocument();

    const timeElement = screen.getByText(/^\w+ \d{1,2}, \d{4}$/);
    expect(timeElement).toHaveAttribute('dateTime');
  });
});
