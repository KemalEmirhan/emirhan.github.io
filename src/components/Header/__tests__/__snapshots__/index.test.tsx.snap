// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header > should render 1`] = `
<div>
  <header
    class="w-full md:max-w-5xl mx-auto"
  >
    <div
      class="grid md:grid-cols-[145px_1fr] items-center py-6 px-4 md:px-0 md:py-8 gap-6"
    >
      <div
        class="hidden md:block border-r border-black border-dashed pr-4 h-full"
      >
        <div
          class="relative w-16 h-16 sm:w-24 sm:h-24 lg:w-32 lg:h-32 rounded-full border border-gray-200 shadow-sm"
        >
          <img
            alt="Emirhan profile picture"
            class="object-cover rounded-full"
            data-nimg="fill"
            decoding="async"
            sizes="100vw"
            src="/_next/image?url=%2Fpp.png&w=3840&q=75"
            srcset="/_next/image?url=%2Fpp.png&w=640&q=75 640w, /_next/image?url=%2Fpp.png&w=750&q=75 750w, /_next/image?url=%2Fpp.png&w=828&q=75 828w, /_next/image?url=%2Fpp.png&w=1080&q=75 1080w, /_next/image?url=%2Fpp.png&w=1200&q=75 1200w, /_next/image?url=%2Fpp.png&w=1920&q=75 1920w, /_next/image?url=%2Fpp.png&w=2048&q=75 2048w, /_next/image?url=%2Fpp.png&w=3840&q=75 3840w"
            style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
          />
        </div>
      </div>
      <div
        class="flex items-center justify-between"
      >
        <h1
          class="text-2xl font-bold"
        >
          emrhnkml.
        </h1>
        <time
          datetime="September 21, 2025"
        >
          September 21, 2025
        </time>
      </div>
    </div>
  </header>
</div>
`;
